doc/antlr-project-testing.md

```diff
@@ -6,52 +6,56 @@ Because ANTLR supports multiple target languages, the unit tests are broken into
-
-```java
-public class ParserExecDescriptors {
-    public static class APlus extends BaseParserTestDescriptor {
-        public String input = "a b c";
-        public String output = "abc\n";
-        public String errors = "";
-        public String startRule = "a";
-        public String grammarName = "T";
-
-        /**
-         grammar T;
-         a : ID+ {
-         <writeln("$text")>
-         };
-         ID : 'a'..'z'+;
-         WS : (' '|'\n') -> skip;
-         */
-        @CommentHasStringValue
-        public String grammar;
-    }
+As of 4.9.4, we use a Java descriptor file held as an [UniversalRuntimeTestDescriptor.java object](https://github.com/antlr/antlr4/blob/master/runtime-testsuite/test/org/antlr/v4/test/runtime/UniversalRuntimeTestDescriptor.java) to represent each runtime test. Each test is described with a text file with various sections and resides in a group directory; see [directories under descriptors dir](https://github.com/antlr/antlr4/blob/master/runtime-testsuite/resources/org/antlr/v4/test/runtime/descriptors). Here is a sample test descriptor:
+
 ```
+[notes]
+This is a regression test for blah blah blah...
+
+[type]
+Parser
+
+[grammar]
+grammar T;
+a : ID* {
+<writeln("$text")>
+};
+ID : 'a'..'z'+;
+WS : (' '|'\n') -> skip;
+
+[start]
+a
 
-The mysterious `@CommentHasStringValue` annotation is a bit of a hack that allows multi-line strings in Java. This kung fu is required so that we can use Java classes rather than StringTemplate group files to specify runtime tests (the legacy system used those and it was hard to get them right). Here are all the [Runtime test descriptors](https://github.com/antlr/antlr4/tree/master/runtime-testsuite/test/org/antlr/v4/test/runtime/descriptors) organized into groups.
+[input]
+a b c
+
+[output]
+"""abc
+"""
+```
 
 The grammars are strings representing StringTemplates (`ST` objects) so `<writeln("$text")>` will get replace when the unit test file is generated (`Test.java`, `Test.cs`, ...). The `writeln` template must be defined per target.  Here are all of the 
-[Target templates for runtime tests](https://github.com/antlr/antlr4/tree/master/runtime-testsuite/resources/org/antlr/v4/test/runtime/templates).
+[Target templates for runtime tests](https://github.com/antlr/antlr4/tree/master/runtime-testsuite/resources/org/antlr/v4/test/runtime/templates).  Use triple-quotes `"""` when whitespace matters (usually input/output sections).
 
 ## Requirements
 
+*out of date, at least for mono*
```

Comment:
Unrelated changes to the pull request. I suggest updating the requirements info in the next PR (replace mono with dotnet).

Reply:
just updating doc as part of broadened PR.